src/platform/packages/shared/response-ops/alerts-table/hooks/use_bulk_update_alert_tags.test.tsx

```diff
@@ -0,0 +1,281 @@
+      add: ['tag1'],
+    });
+
+    await waitFor(() => expect(notifications.toasts.addSuccess).toHaveBeenCalled());
+
+    expect(notifications.toasts.addSuccess).toHaveBeenCalledWith('Updated tags for alert');
+  });
+
+  it('should call onSuccess callback on successful mutation', async () => {
+    http.post.mockResolvedValue('success');
+
+    const { result } = renderHook(
+      () => useBulkUpdateAlertTags({ http, notifications, onSuccess, onError }),
+      { wrapper }
+    );
+
+    result.current.mutate({
+      index: '.alerts-test',
+      alertIds: ['alert-1'],
+      add: ['tag1'],
+    });
+
+    await waitFor(() => expect(onSuccess).toHaveBeenCalled());
+
+    expect(onSuccess).toHaveBeenCalledTimes(1);
+  });
+
+  it('should not call onSuccess callback when onSuccess is undefined', async () => {
+    http.post.mockResolvedValue('success');
+
+    const { result } = renderHook(() => useBulkUpdateAlertTags({ http, notifications }), {
+      wrapper,
+    });
+
+    result.current.mutate({
+      index: '.alerts-test',
+      alertIds: ['alert-1'],
+      add: ['tag1'],
+    });
+
+    await waitFor(() => expect(notifications.toasts.addSuccess).toHaveBeenCalled());
+
+    expect(onSuccess).not.toHaveBeenCalled();
+  });
+
+  it('should display error toast with alert message on failure', async () => {
+    http.post.mockRejectedValue(new Error('API Error'));
+
+    const { result } = renderHook(
+      () => useBulkUpdateAlertTags({ http, notifications, onSuccess, onError }),
+      { wrapper }
+    );
+
+    result.current.mutate({
+      index: '.alerts-test',
+      alertIds: ['alert-1'],
+      add: ['tag1'],
+    });
+
+    await waitFor(() => expect(notifications.toasts.addDanger).toHaveBeenCalled());
+
+    expect(notifications.toasts.addDanger).toHaveBeenCalledWith('Failed to update tags for alert');
+  });
+
+  it('should call onError callback on failed mutation', async () => {
+    http.post.mockRejectedValue(new Error('API Error'));
+
+    const { result } = renderHook(
+      () => useBulkUpdateAlertTags({ http, notifications, onSuccess, onError }),
+      { wrapper }
+    );
+
+    result.current.mutate({
+      index: '.alerts-test',
+      alertIds: ['alert-1'],
+      add: ['tag1'],
+    });
+
+    await waitFor(() => expect(onError).toHaveBeenCalled());
+
+    expect(onError).toHaveBeenCalledTimes(1);
+  });
+
+  it('should not call onError callback when onError is undefined', async () => {
+    http.post.mockRejectedValue(new Error('API Error'));
+
+    const { result } = renderHook(() => useBulkUpdateAlertTags({ http, notifications }), {
+      wrapper,
+    });
+
+    result.current.mutate({
+      index: '.alerts-test',
+      alertIds: ['alert-1'],
+      add: ['tag1'],
+    });
+
+    await waitFor(() => expect(notifications.toasts.addDanger).toHaveBeenCalled());
+
+    expect(onError).not.toHaveBeenCalled();
+  });
+
+  it('should update loading state during mutation', async () => {
```

Comment:
Thank you so much for this great coverage! 

nit: Maybe we can remove this test as it tests React query and there is no custom logic regarding the `isLoading` and `isSuccess` booleans.